core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -567,13 +584,13 @@ class LogManager(logDirs: Array[File],
           info("Created log for partition [%s,%d] in %s with properties {%s}."
             .format(topicPartition.topic,
               topicPartition.partition,
-              dataDir.getAbsolutePath,
+              logDir,
```

Comment:
This is because we receive logDir as a strong from AlterReplicaDirRequest and it is inserted into `preferredLogDirs` as a strong. When we get it from `preferredLogDirs` it will be a strong not a file.

It is a minor change. Alternatively I can insert preferred log directory into that map as a file. I just feel the current approach is simpler.